libs/utils/trace_analysis.py

```diff
@@ -912,54 +810,216 @@ def plotCPUFrequencyResidency(self, cpus=None):
+            yrange = 0.3 * max(6, len(total_time)) * n_cpus
+
+            axes = fig.add_subplot(gs[idx])
+            total_time.plot.barh(ax=axes,
                                  color='g',
                                  legend=False,
-                                 figsize=(16,8),
-                                 fontsize=16);
-            active_time.plot.barh(ax=axis,
+                                 figsize=(16,yrange));
+            active_time.plot.barh(ax=axes,
                                   color='r',
                                   legend=False,
-                                  figsize=(16,8),
-                                  fontsize=16);
-            axis.set_title('CPU{}'.format(cpu))
-            axis.set_xlabel('Time [s]')
-            axis.set_xlim(0, self.x_max)
-
-        if axis:
-            handles, labels = axis.get_legend_handles_labels()
-            plt.figlegend(handles,
-                          labels,
-                          loc='lower right')
-            plt.subplots_adjust(hspace=0.5)
+                                  figsize=(16,yrange));
+
+            axes.set_title('CPU{}'.format(cpu))
+
+            axes.grid(True);
+            if (idx+1 < n_cpus):
+                axes.set_xticklabels([])
+            else:
+                axes.set_xlabel('Time [s]')
+            axis.append(axes)
+
+        # Set same x-axes limit for all plots
+        for ax in axis:
+            ax.set_xlim(0, 1.05 * xmax)
+
+        # Put title on top of the figure. As of now there is no clean way to
+        # make the title appear always in the same position in the figure
+        # because figure heights may vary between different platforms
+        # (different number of OPPs). Hence, we use annotation
+        legend_y = axis[0].get_ylim()[1]
+        axis[0].annotate('OPP Residency Time',
+                         xy=(0, legend_y),
+                         xytext=(-50, 45),
+                         textcoords='offset points',
+                         fontsize=18)
+        axis[0].annotate('GREEN: Total',
+                         xy=(0, legend_y),
+                         xytext=(-50, 25),
+                         textcoords='offset points',
+                         color='g',
+                         fontsize=14)
+        axis[0].annotate('RED: Active',
+                         xy=(0, legend_y),
+                         xytext=(50, 25),
+                         textcoords='offset points',
+                         color='r',
+                         fontsize=14)
+        figname = '{}/{}cpu_freq_residency.png'.format(self.plotsdir,
+                                                       self.prefix)
+        pl.savefig(figname, bbox_inches='tight')
+
+    def _getClusterActiveSignal(self, cluster):
+        """
+        Build a square wave representing the active (i.e. non-idle) cluster
+        time, i.e.:
+            cluster_active[t] == 1 if the CPU is reported to be exited from an
```

Comment:
This is technically incorrect, there is always a task running in every CPU, eventually the idle task.
Let's try to make it less ambiguous by saying something like:
```python
#     cluster_active[t] == 1 if at least one CPU has been reported to be non-idle by CPUFreq at time t
```